lazyflow/operators/ioOperators/opTiffReader.py

```diff
@@ -33,97 +39,63 @@ def __init__(self, *args, **kwargs):
-                dtype_code = self._dtype = tifffile.TIFF_SAMPLE_DTYPES.get(key, None)
-
-            # From the tifffile.TiffPage code:
-            # shaped : tuple of int
-            #     Normalized 5-dimensional shape of the image in IFD:
-            #     0 : separate samplesperpixel or 1.
-            #     1 : imagedepth Z or 1.
-            #     2 : imagelength Y.
-            #     3 : imagewidth X.
-            #     4 : contig samplesperpixel or 1.
-
-            (P, D, Y, X, S) = first_page.shaped
-            assert P == 1, "Don't know how to handle any number of planar samples per pixel except 1 (per page)"
-            assert D == 1, "Don't know how to handle any image depth except 1"
-
-            if S == 1:
-                self._page_shape = (Y, X)
-                self._page_axes = "yx"
-            else:
-                assert shape[-3:] == (Y, X, S)
-                self._page_shape = (Y, X, S)
-                self._page_axes = "yxc"
-                assert "C" not in axes, (
-                    "If channels are in separate pages, then each page can't have multiple channels itself.\n"
-                    "(Don't know how to weave multi-channel pages together.)"
-                )
-
-            self._non_page_shape = shape[: -len(self._page_shape)]
-            assert shape == self._non_page_shape + self._page_shape
-            assert self._non_page_shape or len(pages) == 1
+            dtype_code = series.dtype
 
-            axes = axes.lower().replace("s", "c")
+            # we treat "sample" axis as "channel"
+            axes = axes.replace("s", "c")
             if "i" in axes:
                 for k in "tzc":
                     if k not in axes:
                         axes = axes.replace("i", k)
                         break
                 if "i" in axes:
-                    raise RuntimeError(
-                        "Image has an 'I' axis, and I don't know what it represents. "
-                        "(Separate T,Z,C axes already exist.)"
+                    raise UnsupportedTiffError(
+                        filepath=self._filepath,
+                        details=f"Image has an 'I' axis, and I don't know what it represents (separate T,Z,C axes already exist): {axes}",
                     )
 
+            # tifffile will add potentially multiple "q" axes to data when saving without specifying them
             if "q" in axes:
-                old_axes = axes
                 axes = get_default_axisordering(shape)
-                logger.warning(
-                    f"Unknown axistags detected - assuming default axis order. Guessed {axes} from {old_axes}."
+
+            if any(ax not in "tzyxc" for ax in axes) or len(shape) > 5:
+                raise UnsupportedTiffError(
+                    filepath=self._filepath,
+                    details=f"Don't know how to read TIFF files with more than 5 dimensions (Your image has {len(shape)} dimensions). Axes: {axes}.",
                 )
```

Comment:
Do we know what `i` and `q` axes mean? If yes, it would be great to mention it somewhere in the comments.
```python
if "q" in axes:
    # tifffile will add potentially multiple "q" axes to data when saving without specifying them
    axes = get_default_axisordering(shape)
else:
    # Treat "sample" as "channel".
    # Treat "i" as either "time", "z", or "channel", in that order.
    for old, new in ("sc", "it", "iz", "ic"):
        axes = axes.replace(old, new)

axes_set = set(axes)
# Do the first 2 conditions make sense for TIFFs?.. Could "s" and "c" occur in the same TIFF?
if len(shape) < 2 or len(shape) > 5 or len(axes_set) != len(axes) or axes_set.difference("tzyxc"):
    raise UnsupportedTiffError(
        filepath=self._filepath,
        f"Only 2D-5D TIFFs with unique 'tzyxc' axes are allowed (got {len(shape)}D TIFF with {axes!r} axes)"
    )
```

Reply:
https://github.com/cgohlke/tifffile/blob/master/tifffile/tifffile.py#L17816-L17836

Reply:
and yeah, I think in principle tiff pages could be 1D, good to check that.